Automattic/woocommerce-subscriptions-core
# Display related orders table when viewing the new "Edit Order" page (HPOS enabled stores)

Fixes #192 

## Description



This PR displays the related orders table meta box on the "Edit Order" and "Edit Subscription" when HPOS is enabled with non-HPOS compatibility.

In addition:
- [x] PHPCS formatting has been fixed in related files.
	(mostly moved to #280)
- [x] PHPCS errors/warnings have been addressed in related files.
	(mostly moved to #280)
	- Some low-hanging errors/warnings have been addressed.
- [x] Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0) - yes, 3 hooks added





## How to test this PR





1. Enable the COT feature and turn on COT ([see guide here](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#easy-way)).
2. Go into **WooCommerce > Settings > Advanced > Custom Data Stores**.
3. Make sure "Use the WooCommerce orders tables" is enabled.
4. Purchase a subscription and note the order ID.
5. From WP admin dashboard, go to **WooCommerce > Orders** and view the "Edit Order" page for the order ID noted above.
6. Notice that the Related Orders table meta box is visible and contains a link to the related Subscription.
   You should not see the current order within this table.
	<img width="792" alt="Screenshot 2022-11-08 at 12 57 37" src="https://user-images.githubusercontent.com/3147296/200465243-366995cc-6fe1-4345-b38b-90238d065660.png">	
7. Click on the related Subscription to be taken to the "Edit Subscription" page.
   You should not see the current subscription within this table.
	<img width="795" alt="Screenshot 2022-11-08 at 12 57 24" src="https://user-images.githubusercontent.com/3147296/200465254-3bc410fd-046e-4035-8557-988f917f8f56.png">
9. Notice that the Related Orders table meta box is visible and contains a link to the parent order.
10. Clicking on the parent order will take you to the "Edit Order" page as above.


Recommended compatibility tests (checked = tested by PR author):
- [x] WCPay
- [x] WC Subs
	- [x] with WC Subs activated + HPOS enabled + syncing on, ~the Related Orders table isn't rendered correctly and following PHP Notice occurs~ it uses a newly deprecated hook. Need to update this issue to clarify:
	https://github.com/woocommerce/woocommerce-subscriptions/issues/4417
- [x] WC 6.8.0 – toggling between this subs-core branch & `trunk` results in identical output.
- [x] WC HPOS enabled + syncing off
- [x] WC HPOS enabled + syncing on
- [x] WC HPOS disabled

## Product impact
